lib/spack/spack/test/concretize_requirements.py

```diff
@@ -586,3 +587,101 @@ def test_non_existing_variants_under_all(concretize_scope, mock_packages):
 
     spec = Spec("callpath ^zmpi").concretized()
     assert "~foo" not in spec
+
+
+@pytest.mark.parametrize(
+    "packages_yaml,spec_str,expected_satisfies",
+    [
+        (
+            """\
+    packages:
+      libelf:
+        require:
+        - one_of: ["%clang"]
+          when: "@0.8.13"
+""",
+            "libelf",
+            [("@0.8.13%clang", True), ("%gcc", False)],
+        ),
+        (
+            """\
+    packages:
+      all:
+        compiler: ["gcc", "clang"]
```

Comment:
Yes, in the sense that if the requirement is disregarded we'll be preferring `gcc` instead of `clang`. Otherwise there is the possibility of selecting `clang` accidentally.

Reply:
Done in 57cd6ed